graveler/graveler.go

```diff
@@ -474,17 +476,17 @@ type graveler struct {
 	CommittedManager CommittedManager
 	StagingManager   StagingManager
 	RefManager       RefManager
-	branchLocker     branchLocker
+	branchLocker     *BranchLocker
 	log              logging.Logger
 }
 
-func NewGraveler(committedManager CommittedManager, stagingManager StagingManager, refManager RefManager) *graveler {
+func NewGraveler(db db.Database, committedManager CommittedManager, stagingManager StagingManager, refManager RefManager) *graveler {
```

Comment:
Why not create a branch locker interface and do dependency injection on that as well? to me it feels like a healthier seperation of concerns, the same way we treat all other parts that graveler is composed of.

Reply:
1. No problem - will do.
2. At first I thought graveler will require Postgres to generate unique identifiers per branch and DB will be used for that. But I liked the hash solution better.

Reply:
Moved the implementation to `ref` package and NewGraveler consume BranchLocker interface